### PR TITLE
Setting up project files for DLP

### PR DESCRIPTION
**Pull request by MEseifan**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+      </exclusions>
+    </dependency>
+
+    <dependency>
+      <!--
+        Required by bigtable-hbase-1.x-mapreduce instead of excluded non-shaded version.
+        Shaded library is used to avoid dependency conflicts with Datastore module on profobuf-java dependency.
+        Bigtable requires version 2.x and Datastore module requires 3.x protocol.
+      -->
+      <groupId>org.apache.hbase</groupId>
```
Comment (nitinmotgi) [inline pom.xml]: Why do we hbase ?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+      <version>${dlp.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${guava.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud.bigdataoss</groupId>
+      <artifactId>bigquery-connector</artifactId>
```
Comment (nitinmotgi) [inline pom.xml]: We should remove unused libraries it causes issues with deployment

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+      <artifactId>hbase-shaded-server</artifactId>
+      <version>${hbase-shaded-server.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+      <version>${dropwizard.metrics-core.version}</version>
+    </dependency>
+
+    <dependency>
```
Comment (nitinmotgi) [inline pom.xml]: This whole section has dependencies that make the final jar too big. Could we clean up unused dependencies and keep it to minimum

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+          <instructions>
+            <Embed-Dependency>*;inline=false;scope=compile</Embed-Dependency>
+            <Embed-Transitive>true</Embed-Transitive>
+            <Embed-Directory>lib</Embed-Directory>
+            <!--Only @Plugin classes in the export packages will be included as plugin-->
+            <_exportcontents>
+              io.cdap.plugin.gcp.*;
+              com.google.cloud.hadoop.*;
+              org.apache.spark.streaming.pubsub*;
+              org.apache.hadoop.hbase.mapreduce.*;
```
Comment (nitinmotgi) [inline pom.xml]: Why are we exposing all these classes ? We should remove unwanted stuff from here too

File: src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
+package io.cdap.plugin.gcp.common;
```
Comment (nitinmotgi) [inline src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java:17]: License missing

File: src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
```
Comment (nitinmotgi) [inline src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java:3]: 2019 year

File: src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  protected String project;
+
+  @Name(NAME_SERVICE_ACCOUNT_FILE_PATH)
+  @Description("Path on the local file system of the service account key used "
+    + "for authorization. Can be set to 'auto-detect' when running on a Dataproc cluster. "
+    + "When running on other clusters, the file must be present on every node in the cluster.")
+  @Macro
+  @Nullable
+  protected String serviceFilePath;
+
```
Comment (nitinmotgi) [inline src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java:53]: All public methods should have documentation

**Review by nitinmotgi — CHANGES_REQUESTED** [review]
Minor clean ups required

File: src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java (hunk 6 above)
Comment (MEseifan) [reply src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java:3]: done

File: src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java (hunk 5 above)
Comment (MEseifan) [reply src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java]: done

File: src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java (hunk 7 above)
Comment (MEseifan) [reply src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java:53]: done

File: pom.xml (hunk 1 above)
Comment (MEseifan) [reply pom.xml]: Removed

File: pom.xml (hunk 2 above)
Comment (MEseifan) [reply pom.xml]: Removed more libraries

File: pom.xml (hunk 3 above)
Comment (MEseifan) [reply pom.xml]: Cleaned up

File: pom.xml (hunk 4 above)
Comment (MEseifan) [reply pom.xml]: Done

**Comment by nitinmotgi** [comment]
Checkstyle doesn’t seem to be working - the issues identify in review would have been stopped during build due to checkstyle checks.

File: src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+public class GCPUtils {
+  public static final String CMEK_KEY = "gcp.cmek.key.name";
+
+  public static ServiceAccountCredentials loadServiceAccountCredentials(String path) throws IOException {
+    File credentialsPath = new File(path);
+    try (FileInputStream serviceAccountStream = new FileInputStream(credentialsPath)) {
+      return ServiceAccountCredentials.fromStream(serviceAccountStream);
+    }
+  }
+
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java]: nit: extra

File: src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+package io.cdap.plugin.gcp.common;
+
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * GCP utility class to get service account credentials
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java]: period at the end of javadoc

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+# Singlenode and test data files.
+/templates/
+/data/
+/data-fabric-tests/data/
+
+# ANTLR4
+/core/gen
+*.tokens
+DirectivesLexer.java
```
Comment (CuriousVini) [inline .gitignore]: why do we need these?

File: README.md [hunk 11]
```diff
@@ -1 +1,46 @@
...
+to the users.
+
+## Slack Team
+
+CDAP Users on Slack: [cdap-users team](https://cdap-users.herokuapp.com)
+
+
+# License and Trademarks
+
+Copyright © 2016-2017 Cask Data, Inc.
```
Comment (CuriousVini) [inline README.md]: 2019?

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,523 @@
...
+  ~ the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>io.cdap.plugin</groupId>
+  <artifactId>dlp</artifactId>
+  <version>1.0.0</version>
```
Comment (CuriousVini) [inline pom.xml]: develop branch should have SNAPSHOT version while we are developing the plugin

Comment (CuriousVini) [reply pom.xml]: also we usually create a develop branch and release branch.

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,523 @@
...
+    <guava.version>27.0.1-jre</guava.version>
+    <httpclient.version>4.5.6</httpclient.version>
+    <jackson.core.version>2.8.11.1</jackson.core.version>
+    <junit.version>4.12</junit.version>
+    <slf4j.version>1.7.5</slf4j.version>
+    <spark.version>2.3.1</spark.version>
+
+
+  </properties>
+
```
Comment (CuriousVini) [inline pom.xml:88]: nit: extra line

File: pom.xml [hunk 14]
```diff
@@ -0,0 +1,523 @@
...
+                <id>attach-javadoc</id>
+                <phase>package</phase>
+                <goals>
+                  <goal>jar</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+
+          <!--          <plugin>-->
```
Comment (CuriousVini) [inline pom.xml]: why do we need to keep this commented?

File: pom.xml [hunk 15]
```diff
@@ -0,0 +1,523 @@
...
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>${junit.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-streaming_2.11</artifactId>
```
Comment (CuriousVini) [inline pom.xml]: why is spark streaming dependency needed for transform plugins?

File: pom.xml [hunk 16]
```diff
@@ -0,0 +1,523 @@
...
+      <version>${spark.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_2.11</artifactId>
```
Comment (CuriousVini) [inline pom.xml]: is this needed?

**Review by CuriousVini — COMMENTED** [review]
for pom file, we only include the dependencies that are needed for the plugin. Other dependencies should be removed.

**Comment by MEseifan** [comment]
> Checkstyle doesn’t seem to be working - the issues identify in review would have been stopped during build due to checkstyle checks.

I'll look into this, for now I will manually run checkstyle before commiting

File: src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java (hunk 9 above)
Comment (MEseifan) [reply src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java]: done

File: src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java (hunk 8 above)
Comment (MEseifan) [reply src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java]: done

File: README.md (hunk 11 above)
Comment (MEseifan) [reply README.md]: done

File: .gitignore (hunk 10 above)
Comment (MEseifan) [reply .gitignore]: Not needed, I copied this file from the google-cloud repo. I removed it

File: pom.xml (hunk 12 above)
Comment (MEseifan) [reply pom.xml]: Done. I created a develop branch and changed the target of this PR to the develop branch

File: pom.xml (hunk 13 above)
Comment (MEseifan) [reply pom.xml:88]: dome

File: pom.xml (hunk 14 above)
Comment (MEseifan) [reply pom.xml]: Meant to remove it, done

File: pom.xml (hunk 15 above)
Comment (MEseifan) [reply pom.xml]: Removed

File: pom.xml (hunk 16 above)
Comment (MEseifan) [reply pom.xml]: Dont think so, removed it

File: checkstyle.xml [hunk 17]
```diff
@@ -0,0 +1,405 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright © 2014-2015 Cask Data, Inc.
```
Comment (CuriousVini) [inline checkstyle.xml]: should be 2019

File: pom.xml [hunk 18]
```diff
@@ -0,0 +1,429 @@
...
+      <url>https://oss.sonatype.org/service/local/staging/deploy/maven2</url>
+    </repository>
+    <snapshotRepository>
+      <id>sonatype.snapshots</id>
+      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+    </snapshotRepository>
+  </distributionManagement>
+
+  <properties>
+    <dlp.version>0.111.0-beta</dlp.version>
```
Comment (CuriousVini) [inline pom.xml:73]: do they have released version? We should not rely on beta version

File: pom.xml [hunk 19]
```diff
@@ -0,0 +1,429 @@
...
+    <jee.version>7</jee.version>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <avro.version>1.8.2</avro.version>
+    <commons.codec.version>1.4</commons.codec.version>
+    <cdap.version>6.1.0-SNAPSHOT</cdap.version>
```
Comment (CuriousVini) [inline pom.xml]: this should be 6.1.1 since its not compatible with 6.1.0-SNAPSHOT

File: suppressions.xml [hunk 20]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
+
+<!--
+  Copyright © 2014-2015 Cask Data, Inc.
```
Comment (CuriousVini) [inline suppressions.xml]: 2019

File: pom.xml [hunk 21]
```diff
@@ -0,0 +1,429 @@
...
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>io.cdap</groupId>
+        <artifactId>cdap-maven-plugin</artifactId>
+        <version>1.1.0</version>
+        <configuration>
+          <cdapArtifacts>
+            <parent>system:cdap-data-pipeline[6.1.0-SNAPSHOT,7.0.0-SNAPSHOT)
```
Comment (CuriousVini) [inline pom.xml]: lower bound should be `6.1.1`

**Review by CuriousVini — APPROVED** [review]
Also please include jira number in the commit. 
fix the comments, squash and merge

**Comment by CuriousVini** [comment]
> > Checkstyle doesn’t seem to be working - the issues identify in review would have been stopped during build due to checkstyle checks.
> 
> I'll look into this, for now I will manually run checkstyle before commiting

Does this work?

File: checkstyle.xml (hunk 17 above)
Comment (MEseifan) [reply checkstyle.xml]: done

File: pom.xml (hunk 19 above)
Comment (MEseifan) [reply pom.xml]: Done

File: pom.xml (hunk 21 above)
Comment (MEseifan) [reply pom.xml]: Done

File: suppressions.xml (hunk 20 above)
Comment (MEseifan) [reply suppressions.xml]: done

File: pom.xml (hunk 18 above)
Comment (MEseifan) [reply pom.xml:73]: There is no release version, they [are all](https://mvnrepository.com/artifact/com.google.cloud/google-cloud-dlp) tagged with beta.

**Review by MEseifan — COMMENTED** [review]
> > > Checkstyle doesn’t seem to be working - the issues identify in review would have been stopped during build due to checkstyle checks.
> > 
> > 
> > I'll look into this, for now I will manually run checkstyle before commiting
> 
> Does this work?

Yes, checkstyle runs with the build. This branch still needs to be configured to trigger builds and block PRs if builds fail. I think an admin needs to configure that?

**Review by CuriousVini — APPROVED** [review]
lgtm

**Pull request merged**